### PR TITLE
remove backcompat handling of 6.2.x versions

### PR DESCRIPTION
**Pull request by talevy**
relates to refactoring initiative #41164.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerTests.java [hunk 1]
```diff
@@ -68,14 +68,16 @@ public void testThatInstancesAreCachedAndReused() {
...
+        Version versionB = randomValueOtherThan(versionA, () -> randomVersion(random()));
+        assertSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(versionA),
+                PreBuiltAnalyzers.STANDARD.getAnalyzer(versionA));
+        assertNotSame(PreBuiltAnalyzers.STANDARD.getAnalyzer(versionA),
+                PreBuiltAnalyzers.STANDARD.getAnalyzer(versionB));
 
         // Same Lucene version should be cached:
-        assertSame(PreBuiltAnalyzers.STOP.getAnalyzer(Version.V_6_2_1),
-            PreBuiltAnalyzers.STOP.getAnalyzer(Version.V_6_2_2));
+        assertSame(PreBuiltAnalyzers.STOP.getAnalyzer(Version.V_7_2_0),
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerTests.java]: This is going to eventually break when 8.0 has lucene upgraded. I think you could do this instead with 2 hardcoded different versions that are old, eg `Version.fromString("5.0.0")` and `Version.fromString("5.0.1")`, which will use the oldest version of lucene we use in Version.java.

File: server/src/test/java/org/elasticsearch/search/slice/SliceBuilderTests.java [hunk 2]
```diff
@@ -461,7 +462,7 @@ public void testSerializationBackcompat() throws IOException {
 
         SliceBuilder copy62 = copyWriteable(sliceBuilder,
                 new NamedWriteableRegistry(Collections.emptyList()),
-                SliceBuilder::new, Version.V_6_2_0);
+                SliceBuilder::new, VersionUtils.getPreviousVersion(Version.V_6_3_0));
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/search/slice/SliceBuilderTests.java]: Isn't this just removed in your 6.3 removal PR?

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerTests.java (hunk 1 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerTests.java]: I see. good catch!

File: server/src/test/java/org/elasticsearch/search/slice/SliceBuilderTests.java (hunk 2 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/search/slice/SliceBuilderTests.java]: let it be that PR's worry :)

**Comment by talevy** [comment]
Hey @rjernst, I've made some changes since your last review due to failing tests.

I went ahead and removed some 6.3 refs as well since they were failing and will be gone anyways

mind doing a quick pass again? thanks!

File: server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java [hunk 3]
```diff
@@ -197,8 +197,8 @@ public void testLuceneVersionOnUnknownVersions() {
                 version.luceneVersion);
 
         // between two known versions, should use the lucene version of the previous version
-        version = Version.fromString("6.2.50");
-        assertEquals(VersionUtils.getPreviousVersion(Version.V_6_2_4).luceneVersion, version.luceneVersion);
+        version = Version.fromString("8.0.50");
+        assertEquals(Version.V_8_0_0.luceneVersion, version.luceneVersion);
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java]: Can we make this not depend on any constants? Perhaps use `VersionUtils.getPreviousVersion(Version.CURRENT).toString() + "1"`, so the bugfix portion will be past any known versions, but still before current.

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java [hunk 4]
```diff
@@ -197,15 +197,15 @@ public void testLuceneVersionOnUnknownVersions() {
                 version.luceneVersion);
```
Comment (talevy) [inline server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java]: Hi @jpountz, Do you feel that the other tests here cover what this test against 6.x is doing?

**Comment by talevy** [comment]
run elasticsearch-ci/1
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro

File: server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java (hunk 4 above)
Comment (jpountz) [reply server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java]: +1

**Review by jpountz — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java (hunk 4 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/common/lucene/uid/VersionsTests.java]: thanks Adrien!

**Pull request merged**